File: src/Python 3/Prerequisites/Prerequisites.py
while True:
    line = input()
    if line == "0":
        break
    k, m = map(int, line.split())
    courses = set()
    arr = [int(i) for i in input().split()]
    for course in arr:
        courses.add(course)
    check = True
    count = 0
    for i in range(m):
        if check == False:
            break
        count += 1
        arr = [int(i) for i in input().split()]
        c = arr[0]
        r = arr[1]
        arr = arr[2:]
        hasMet = 0
        for j in arr:
            if j in courses:
                hasMet += 1
        check = hasMet >= r
    for i in range(count, m):
        input()
    print("yes" if check else "no")

File: src/Java/Prerequisites/Prerequisites.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.io.IOException;
import java.util.Set;
import java.util.HashSet;

public class Prerequisites {
    public static void main(String[] args) throws IOException {
        BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter dc = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));

        while (true) {
            String line = sc.readLine();
            if (line.equals("0")) {
                break;
            }
            String[] temp = line.split(" ");
            int k = Integer.parseInt(temp[0]);
            int m = Integer.parseInt(temp[1]);

            Set<Integer> set = new HashSet<>();
            String[] courses = sc.readLine().split(" ");
            for (int i = 0; i < k; i++) {
                set.add(Integer.parseInt(courses[i]));
            }

            boolean check = true;
            int i = 0;
            for (; i < m && check; i++) {
                String[] cat = sc.readLine().split(" ");
                int c = Integer.parseInt(cat[0]);
                int r = Integer.parseInt(cat[1]);
                int hasMet = 0;
                for (int j = 2; j < c + 2; j++) {
                    int course = Integer.parseInt(cat[j]);
                    if (set.contains(course)) {
                        hasMet++;
                    }
                }
                check = hasMet >= r;
            }
            for (; i < m; i++) {
                sc.readLine();
            }
            dc.println(check ? "yes" : "no");
        }
        dc.close();
        sc.close();
    }
}